### PR TITLE
Add webnn softmax tests

### PR DESCRIPTION
**Pull request by BruceDai**
@fdwr @Honry @huningxin @dontcallmedom PTAL, thanks.

### CODE REVIEW

File: webnn/resources/utils.js [hunk 1]
```diff
@@ -1,61 +1,350 @@
...
+
+/**
+ * Get ULP tolerance of softmax operation.
+ * @param {Object} resources - Resources used for building a graph
+ * @returns {Number} A tolerance number
+ */
+const getSoftmaxPrecisionTolerance = (resources) => {
+  // Compute the softmax values of the 2-D input tensor along axis 1.
+  const inputShape = resources.inputs[Object.keys(resources.inputs)[0]].shape;
+  const tolerance = inputShape[1] * 2 + 3;
```
Comment (fdwr) [inline webnn/resources/utils.js:71]: `IEPOE*3+3 ULP` Sorry for the typo. I updated it here: https://github.com/webmachinelearning/webnn/issues/265. The subtraction, exponentiation, and addition all contribute to loss in `expₑ(a - reducemax(A, axes)) / reducesum(expₑ(A - reducemax(A, axes)), axis)`.

**Review by fdwr — COMMENTED** [review]
One comment. Otherwise signed off.

File: webnn/resources/utils.js (hunk 1 above)
Comment (BruceDai) [reply webnn/resources/utils.js]: I've updated, thanks @fdwr

**Review by fdwr — APPROVED** [review]

**Comment by BruceDai** [comment]
@Honry All checks passed, PTAL, thanks.

**Review by Honry — APPROVED** [review]
LGTM, thanks!

**Pull request merged**